### PR TITLE
Support text files

### PR DESCRIPTION
**Pull request by tobiasschweizer**
This PR adds support for text files in the Lua upload script to be used with Knora.

### CODE REVIEW

File: scripts/test_mediatype.lua [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        expected = false
+    }
+}
+
+
+for i, test_data_item in ipairs(file_extension_test_data) do
+    local check = check_file_extension(test_data_item.received1, test_data_item.received2)
+
+    if (check ~= test_data_item.expected) then
+        send_error(400, "Could not correctly check consistency between mimetype and file extension for " .. test_data_item.received1 .. ", "..  test_data_item.received2)
```
Comment (Vijeinath) [inline scripts/test_mediatype.lua]: Make the

Comment (tobiasschweizer) [reply scripts/test_mediatype.lua]: done in https://github.com/dhlab-basel/Sipi/pull/234/commits/ba9a36b031cfd7e3ff21399a22c16ca56b511fbb

**Comment by loicjaouen** [comment]
sorry to ask, what does sipi do with these text files? is it for meta data, for storing text?

**Comment by tobiasschweizer** [comment]
I use them to store XSL transformation and text files in Sipi that are referred to be Knora file values.

**Comment by loicjaouen** [comment]
cool! would you share a code sample that sending/retrieving a text file (an http request as a curl command or something like that)?

**Comment by tobiasschweizer** [comment]
see "create a TextRepresentation of type XSLTransformation and refer to it in a mapping" in `org/knora/webapi/e2e/v1/KnoraSipiIntegrationV1ITSpec.scala`. When you create a `http://www.knora.org/ontology/knora-base#XSLTransformation` and you query the knora resource, it gives you the Sipi path.

**Comment by loicjaouen** [comment]
many thanks!

**Comment by gfoo** [comment]
is it related to PDF support ? (see comments in #222)

**Comment by tobiasschweizer** [comment]
yes

please note that this PR is already closed and merged

**Pull request merged**